lib/src/components/NxTable/__tests__/NxTable.test.tsx

```diff
@@ -4,163 +4,446 @@
-      <NxTable>
-        <NxTableHead>
-          <NxTableRow>
-            <NxTableCell>Foo</NxTableCell>
-            <NxTableCell>Bar</NxTableCell>
-          </NxTableRow>
-        </NxTableHead>
-        <NxTableBody error="Error!"/>
-      </NxTable>,
-      { attachTo: mountContainer }
-    );
+          expect(screen.getByRole('button')).toHaveAccessibleName('dolphin');
+        });
+      });
+
+      describe('when isHeader', function() {
+        it('renders a header cell with the text "Select Row"', function() {
+          const component = renderColumnHeader({
+            chevron: true,
+            sortDir: 'asc',
+            children: <span>foo</span>
+          })!;
+
+          expect(component).toHaveTextContent('Select Row');
+        });
+      });
+    });
+
+    describe('when clickable and the rowBtnIcon is set', function() {
+      const renderTableCell = (cellProps?: NxTableCellProps, rowProps?: NxTableRowProps, isHeader?: boolean) => {
+        const children = (
+          <NxTable.Row isClickable={true} { ...rowProps }>
+            <NxTable.Cell>Foo</NxTable.Cell>
+            <NxTable.Cell>Bar</NxTable.Cell>
+            <NxTable.Cell { ...cellProps }>Boo</NxTable.Cell>
+          </NxTable.Row>
+        );
+
+        return render(
+          <NxTable>
+            { isHeader ? <NxTable.Head>{children}</NxTable.Head>
+            : <NxTable.Body>{children}</NxTable.Body>
+            }
+          </NxTable>
+        ).getAllByRole(isHeader ? 'columnheader' : 'cell')[2] as HTMLElement;
+      };
+
+      describe('when isHeader', function() {
+        it('renders a header cell with the text "Select Row"', function() {
+          const component = renderTableCell({
+            rowBtnIcon: faEdit,
+            sortDir: 'asc'
+          }, undefined, true)!;
+
+          expect(component).toHaveTextContent('Select Row');
+        });
+      });
+
+      describe('when not isHeader', function() {
+        it('does not show button when rowBtnIcon is undefined', function() {
+          expect(within(renderTableCell({ rowBtnIcon: undefined })).queryByRole('button')).not.toBeInTheDocument();
+        });
+
+        it('does not show button when rowBtnIcon is null', function() {
+          expect(within(renderTableCell({ rowBtnIcon: null })).queryByRole('button')).not.toBeInTheDocument();
+        });
+
+        it('show button when rowBtnIcon faEdit is set', function() {
+          expect(within(renderTableCell({ rowBtnIcon: faEdit })).getByRole('button')).toBeInTheDocument();
+        });
+
+        it('ignores the children and sort settings and renders a button instead',
+            function() {
+              const cell = renderTableCell({
+                rowBtnIcon: faEdit,
+                sortDir: 'asc'
+              });
+
+              const button = within(cell).getByRole('button');
+              expect(button).toBeInTheDocument();
+              expect(cell).not.toContainHTML('<span>foo</span>');
```

Comment:
I still don't really know what this line is trying to test.  For one thing, doesn't the table cell under test contain "Boo" in the JSX, not "foo"? But additionally the span seems like an implementation detail.  If the goal is to assert that the cell does not contain its JSX children in this case, I'd just use `.not.toHaveTextContent` for that.  On the other hand if there's something I'm missing about what this line is trying to test, it needs a comment explaining it.

Reply:
> I still don't really know what this line is trying to test. For one thing, doesn't the table cell under test contain "Boo" in the JSX, not "foo"? But additionally the span seems like an implementation detail. If the goal is to assert that the cell does not contain its JSX children in this case, I'd just use `.not.toHaveTextContent` for that. On the other hand if there's something I'm missing about what this line is trying to test, it needs a comment explaining it.

Yeah I wanted to assert that the cell does not contain its JSX children:
Thanks for the suggestion, updated the test name and impl here:
2af1198e4537b02a8ad8237f0e06667307bf85c9

Reply:
> Yeah I wanted to assert that the cell does not contain its JSX children:
Thanks for the suggestion, updated the test name and impl here:

But you're still checking for "foo" when the test setup has "Boo"

Reply:
> > Yeah I wanted to assert that the cell does not contain its JSX children:
> > Thanks for the suggestion, updated the test name and impl here:
> 
> But you're still checking for "foo" when the test setup has "Boo"

I forgot I changed it to that: 5752a72adee7882dfcff10294deba663122e2d37